runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/DoFnOperator.java

```diff
@@ -905,6 +928,19 @@ public final void notifyCheckpointComplete(long checkpointId) throws Exception {
       bufferingDoFnRunner.checkpointCompleted(checkpointId);
     }
 
+    List<InMemoryBundleFinalizer.Finalization> finalizations =
+        pendingFinalizations.get(checkpointId);
+    if (finalizations != null) {
+      // remove old finalizations except for the current one
+      pendingFinalizations.clear();
+      pendingFinalizations.put(checkpointId, finalizations);
```

Comment:
Why is this one pending when we acknowledge it below?

Reply:
I was basing my logic off of UnboundedSourceWrapper which seemed flawed but didn't know enough as to why it was done that way there.